src/lib/core/quiz.ts

```diff
@@ -22,3 +29,131 @@ export const createQuiz = async (quizData: ICreateQuizProps) => {
   const quiz = await prisma.quiz.create({data: quizData})
   return quiz
 }
+
+export const createQuizSolution = async (
+  solutionData: ICreateQuizSolutionProps,
+) => {
+  if (!solutionData.quizId) {
+    throw new Error('missing quiz id')
+  } else if (!solutionData.code) {
+    throw new Error('missing code')
+  } else if (!solutionData.sequence) {
+    throw new Error('missing sequence')
+  } else if (!solutionData.importDirectives) {
+    throw new Error('missing import directives')
+  } else if (!solutionData.testRunner) {
+    throw new Error('missing test runner')
+  }
+  const solution = await prisma.solution.create({
+    data: solutionData,
+  })
+  return solution
+}
+
+export const createQuizTestCases = async (
+  testCaseData: ICreateQuizTestCaseProps[],
+) => {
+  if (!testCaseData || testCaseData.length === 0) {
+    throw new Error('test case not found')
+  }
+  let missingField
+  const hasAllProps = testCaseData.every((testCase) => {
+    return ['solutionId', 'input', 'output', 'sequence'].every((prop) => {
+      if (!testCase.hasOwnProperty(prop)) {
```

Comment:
Use Object.hasOwn() instead of object.hasOwnProperty()

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn